dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -196,12 +197,18 @@ func startSyncWriter(queue util.Queue) util.Queue {
 }
 
 func (cw *ClientWriter) sendSuccessPiece(piece *Piece, cost time.Duration) {
-	cw.api.ReportPiece(piece.SuperNode, &types.ReportPieceRequest{
+	resp, e := cw.api.ReportPiece(piece.SuperNode, &types.ReportPieceRequest{
 		TaskID:     piece.TaskID,
 		Cid:        cw.cfg.RV.Cid,
 		DstCid:     piece.DstCid,
 		PieceRange: piece.Range,
 	})
+	if e != nil {
+		logrus.Errorf("sendSuccessPiece error:%v", e)
+	}
+	if resp.Code != constants.CodeGetPieceReport {
+		logrus.Errorf("send success piece report to supernode failed")
```

Comment:
logrus.Errorf("failed to report piece to supernode: api response code is %d not equal to %d, resp.Code, constants.CodeGetPieceReport")

When we are reporting an error to one user, we should provide as many details as possible. Only by that can user get more accurate information of the actual state. @yunfeiyangbuaa 
And it will reduce more effort you paid in debugging.

Reply:
Done